source/base/utilities.cc

```diff
@@ -123,6 +124,37 @@ namespace Utilities
     return std::string( input, start_idx, end_idx+1-start_idx);
   }
 
+
+
+  std::string
+  trim_whitespace(const std::string &input)
+  {
+    std::string::size_type left = 0;
+    std::string::size_type right = input.size() - 1;
+
+    std::locale current_locale;
+
+    for (; left < input.size(); ++left)
+      {
+        if (!std::isspace(input[left], current_locale))
```

Comment:
Yes, but my intent was to use the current locale, which may define things as whitespace which the current C locale does not. The 'C locale' is really a POSIX standard for character definitions [1]. I believe "the currently installed C locale" may be different from the global locale C++ uses for this reason. Internationalization is hard and I might still not understand how this whole thing works.

Regardless, I think it would be best to ignore the current locale alltogether and do what you recommend (it is not like this bit of code can handle generic unicode spaces anyway). 

[1] http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap07.html